connector/github/github.go

```diff
@@ -303,11 +358,82 @@ func (c *githubConnector) user(ctx context.Context, client *http.Client) (user,
 	return u, nil
 }
 
+// userInOrgTeams first queries the GitHub API for a users' org membership, and
+// if user is an org member, queries the API for team membership within the org.
+// userInOrgTeams will return false if user is not in the org, and an empty
+// slice if user is in the org but not any teams in the org
+//
+// The HTTP passed client is expected to be constructed by the golang.org/x/oauth2 package,
+// which inserts a bearer token as part of the request.
+func (c *githubConnector) userInOrgTeams(ctx context.Context, client *http.Client, userName string, org Org) ([]string, bool, error) {
+	if inOrg, err := c.userInOrg(ctx, client, userName, org.Name); err != nil || !inOrg {
+		return nil, false, err
+	}
+
+	userTeams, err := c.teams(ctx, client, org.Name)
+	if err != nil {
+		return nil, true, err
+	}
+	// Return all teams user is a member of if no team restrictions
+	if len(org.Teams) == 0 {
+		return userTeams, true, nil
+	}
+
+	wantedTeams := make(map[string]struct{})
+	for _, team := range org.Teams {
+		if _, ok := wantedTeams[team]; !ok {
+			wantedTeams[team] = struct{}{}
+		}
+	}
+	teams := make([]string, 0)
+	for _, team := range userTeams {
+		if _, ok := wantedTeams[team]; ok {
+			teams = append(teams, team)
+		}
+	}
+
+	return teams, true, nil
+}
+
+// userInOrg queries the GitHub API for a users' org membership.
+//
+// The HTTP passed client is expected to be constructed by the golang.org/x/oauth2 package,
+// which inserts a bearer token as part of the request.
+func (c *githubConnector) userInOrg(ctx context.Context, client *http.Client, userName, orgName string) (bool, error) {
+	// requester == user, so GET-ing this endpoint should return 404/302 if user
+	// is not a member
+	//
+	// https://developer.github.com/v3/orgs/members/#check-membership
+	apiURL := fmt.Sprintf("%s/orgs/%s/members/%s", c.apiURL, orgName, userName)
+
+	req, err := http.NewRequest("GET", apiURL, nil)
+
+	if err != nil {
+		return false, fmt.Errorf("github: new req: %v", err)
+	}
+	req = req.WithContext(ctx)
+	resp, err := client.Do(req)
+	if err != nil {
+		return false, fmt.Errorf("github: get teams: %v", err)
+	}
+	defer resp.Body.Close()
+
+	err = nil
+	switch resp.StatusCode {
+	case http.StatusNoContent, http.StatusFound, http.StatusNotFound:
```

Comment:
Status(Not)Found mean the same thing in our context (requester == user), so I'll return a 'user not in org' error